js/src/instructions/withdraw.ts

```diff
@@ -0,0 +1,102 @@
+import * as BufferLayout from '@solana/buffer-layout';
+import { TOKEN_PROGRAM_ID } from '@solana/spl-token';
+import {
+  PublicKey,
+  StakeProgram,
+  SystemProgram,
+  SYSVAR_CLOCK_PUBKEY,
+  TransactionInstruction,
+} from '@solana/web3.js';
+import BN from 'bn.js';
+import type { Lamports, ProgramAddresses, Snapshot } from '../types';
+import {
+  findAuthorityProgramAddress,
+  getHeaviestValidatorStakeAccount,
+} from '../utils';
+
+export const getWithdrawInstruction = async (
+  snapshot: Snapshot,
+  programAddresses: ProgramAddresses,
+  stSolAccountOwnerAddress: PublicKey,
+  senderStSolAccountAddress: PublicKey,
+  recipientStakeAccountAddress: PublicKey,
+  amount: Lamports
+) => {
+  const stakeAuthorityAddress = await findAuthorityProgramAddress(
+    programAddresses,
+    'stake_authority'
+  );
+  const heaviestValidatorStakeAccount =
+    getHeaviestValidatorStakeAccount(snapshot);
+
+  if (amount.lamports.lte(snapshot.stakeAccountRentExemptionBalance.lamports)) {
+    throw new Error('Amount must be greater than the rent exemption balance');
+  }
+
+  const maxWithdrawAmount = heaviestValidatorStakeAccount.balance.lamports
```

Comment:
To clarify that the amount is in SOL, we can wrap the expression on the right-hand side in `Lamports()`. These wrapper types are not just useful for users, also for us!